dap4/d4tests/src/test/java/dap4/test/TestDAP4Client.java

```diff
@@ -0,0 +1,406 @@
+  static final String[] XFAIL_TESTS = {};
+
+  static boolean isXfailTest(String t) {
+    for (String s : XFAIL_TESTS) {
+      if (s.equals(t))
+        return true;
+    }
+    return false;
+  }
+
+  //////////////////////////////////////////////////
+  // Type Declarations
+
+  static class Server {
+    public static String SERVLET = "dts";
+    public String ip;
+    public String port;
+
+    public Server(String ip, String port) {
+      this.ip = ip;
+      this.port = port;
+    }
+
+    public String getURL() {
+      StringBuilder buf = new StringBuilder();
+      buf.append("http://");
+      buf.append(this.ip);
+      if (port != null) {
+        buf.append(":");
+        buf.append(this.port);
+      }
+      return buf.toString();
+    }
+
+    // Return a URL for testing if server is up/down
+    public String getTestURL() {
+      StringBuilder baseurl = new StringBuilder().append(getURL());
+      baseurl.append("/");
+      baseurl.append(SERVLET);
+      return baseurl.toString();
+    }
+
+  }
+
+  static class ClientTest {
+    static String root = null;
+    static String server = null;
+    static String servlet = null;
+    static int counter = 0;
+
+    boolean checksumming = true;
+    boolean xfail = false;
+    boolean headeronly = false;
+
+    String title;
+    String dataset; // path minus the server url part.
+    String baselinepath;
+    String constraint;
+    int id;
+
+    ClientTest(String dataset) {
+      this(0, dataset, null);
+    }
+
+    ClientTest(int id, String datasetpath, String constraint) {
+      if (constraint == null)
+        constraint = "";
+      // Break off the final file set name
+      int index = datasetpath.lastIndexOf('/');
+      this.dataset = datasetpath.substring(index + 1, datasetpath.length());
+      this.title = this.dataset;
+      this.id = id;
+      this.constraint = (constraint.length() == 0 ? null : constraint);
+      this.baselinepath = root + "/" + BASELINEDIR + "/" + dataset;
+      if (this.constraint != null)
+        this.baselinepath += ("." + String.valueOf(this.id));
+    }
+
+    public ClientTest nochecksum() {
+      this.checksumming = false;
+      return this;
+    }
+
+    public ClientTest xfail() {
+      this.xfail = true;
+      return this;
+    }
+
+    public ClientTest headeronly() {
+      this.headeronly = true;
+      return this;
+    }
+
+    String makeurl() {
+      String url = this.server + "/" + this.servlet + "/" + this.dataset;
+      if (constraint != null)
+        url += ("?" + constraint);
+      url += "#dap4";
+      return url;
+    }
+
+    public String toString() {
+      return dataset;
+    }
+  }
+
+  //////////////////////////////////////////////////
+  // Class variables
+
+  // Order is important; testing reachability is in the order listed
+  static List<Server> SERVERS;
+
+  static {
+    SERVERS = new ArrayList<>();
+    SERVERS.add(new Server("149.165.169.123", "8080"));
```

Comment:
In no way should this be merged with this IP address here. We are not doing more of this.

Reply:
Ok, test removed until we can resolve this.